src/agent/src/image_rpc.rs

```diff
@@ -119,29 +363,48 @@ impl ImageService {
     }
 
     async fn pull_image(&self, req: &image::PullImageRequest) -> Result<String> {
+        env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", KEYPROVIDER_PATH);
+
         let image = req.get_image();
-        let mut cid = req.get_container_id();
+        let mut cid = req.get_container_id().to_string();
 
         if cid.is_empty() {
             let v: Vec<&str> = image.rsplit('/').collect();
             if !v[0].is_empty() {
-                cid = v[0]
+                cid = v[0].replace(":", "_");
             } else {
                 return Err(anyhow!("Invalid image name. {}", image));
             }
         } else {
-            verify_cid(cid)?;
+            verify_cid(&cid)?;
         }
 
         let source_creds = (!req.get_source_creds().is_empty()).then(|| req.get_source_creds());
 
-        // Read the policy path from the agent config
-        let config_policy_path = &AGENT_CONFIG.read().await.container_policy_path;
-        let policy_path = (!config_policy_path.is_empty()).then(|| config_policy_path);
-        info!(sl!(), "Using container policy_path: {:?}...", &policy_path);
+        if Path::new(SKOPEO_PATH).exists() {
+            // Read the policy path from the agent config
+            let config_policy_path = &AGENT_CONFIG.read().await.container_policy_path;
+            let policy_path = (!config_policy_path.is_empty()).then(|| config_policy_path);
+            info!(sl!(), "Using container policy_path: {:?}...", &policy_path);
+
+            Self::pull_image_from_registry(image, &cid, &source_creds, &policy_path)?;
+        } else {
+            let image = image.to_string();
+            let cid = cid.to_string();
+            let source_creds =
+                (!req.get_source_creds().is_empty()).then(|| req.get_source_creds().to_string());
+            // ocicrypt-rs keyprovider module will create a new runtime to talk with
+            // attestation agent, to avoid startup a runtime within a runtime, we
+            // spawn a new thread here.
+            tokio::task::spawn_blocking(move || {
+                Self::pull_image_with_oci_distribution(&image, &cid, &source_creds)
+                    .map_err(|err| println!("{:?}", err))
```

Comment:
Why the call to `println!()` and not `info!()` or similar?

Reply:
thanks, will fix it